skynet/modules/ttt/rag/stores/s3.py

```diff
@@ -21,6 +23,16 @@ async def files_aiter():
 class RagS3:
     def __init__(self):
         self.s3 = S3()
+        self.listen_task = None
+
+    def __del__(self):
```

Comment:
In general is not great to rely on `__del__` because it can be non-determinisstic when it runs, as it relies on GC.

An explicit `close()` or `dispose()` method would be better. Do you think we could do that?

Reply:
haha, that's how I initially implemented it but then I thought of the destructor, will. revert the the OG